src/App.js

```diff
@@ -1,23 +1,71 @@
-import logo from './logo.svg';
+// import logo from './logo.svg';
+import { useState } from 'react';
 import './App.css';
+import contacts from "./contacts.json";
 
 function App() {
+
+  const firstFive = contacts.slice(0, 5);
+  const[contactsArray, setContactsArray] = useState(firstFive)
+
+  const randomizeContacts = () => {
+    const shuffledContacts = contacts.sort(() => 0.5 - Math.random());
+    const randomFive = shuffledContacts.slice(0, 5);
+    setContactsArray(randomFive);
+  };
+
+  const sortByPopularity = () => {
+    const sortedContacts = [...contactsArray].sort(
+      (a, b) => b.popularity - a.popularity
+    );
+    setContactsArray(sortedContacts);
+  };
+
+  const sortByName = () => {
+    const sortedContacts = [...contactsArray].sort((a, b) =>
+      a.name.localeCompare(b.name)
+    );
+    setContactsArray(sortedContacts);
+  };
+
+  const deleteContact = (id) => {
+    const newListContact = contactsArray.filter(element => element.id !== id)
+    setContactsArray(newListContact);
+  }
+
   return (
+
+
     <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
+      <h1>IconContacts</h1>
+      <button onClick={randomizeContacts}>Random Contacts</button>
+      <button onClick={sortByPopularity}>Sort By Popularity</button>
+      <button onClick={sortByName}>Sort By Name</button>
+      <table className="Header-Table">
+        <tr className="Header">
+          <th>Picture</th>
+          <th>Name</th>
+          <th>Popularity</th>
+          <th>Won Oscar</th>
+          <th>Won Emmy</th>
+          <th>Actions</th>
+        </tr>
+      </table>
+      
+      {contactsArray.map(element => {
```

Comment:
You probably just wanted to add a row to the previous table, and also, remember that when we map through a list we should add a key that would be a unique identifier for the JSX we are rendering